src/domain/repositories/DataSetsRepository.ts

```diff
@@ -1,7 +1,8 @@
 import { DataSet } from "domain/entities/DataSet";
-import { Id } from "types/d2-api";
+import { Id, MetadataResponse } from "types/d2-api";
 
 export interface DataSetsRepository {
     get(ids: Id[]): Promise<Record<Id, DataSet>>;
+    post(data: object, options: object): Promise<MetadataResponse>;
```

Comment:
`MetadataResponse` is not a domain entity, it's an implementation detail of d2-api. Returns the most simple value needed by your usecase.

Reply:
Changed it to return the post status.